CyclopsNuclearReactor/CyNukeReactorMono.cs

```diff
@@ -0,0 +1,294 @@
+﻿namespace CyclopsNuclearReactor
+{
+    using Common;
+    using MoreCyclopsUpgrades.Managers;
+    using ProtoBuf;
+    using System;
+    using System.Reflection;
+    using UnityEngine;
+
+    [ProtoContract]
+    internal class CyNukeReactorMono : HandTarget, IHandTarget, IProtoEventListener, IProtoTreeEventListener
+    {
+        internal static readonly string[] SlotNames = { "CyNuk1", "CyNuk2", "CyNuk3", "CyNuk4" };
+        internal const int MaxSlots = 4;
+        internal const float InitialReactorRodCharge = 10000f; // Half of what the Base Nuclear Reactor provides
+        internal const float PowerMultiplier = 4.1f; // Rounded down from what the Base Nuclear Reactor provides
+
+        public SubRoot ParentCyclops;
+        public CyNukeChargeManager Manager;
+
+        private Equipment _rodSlots;
+        private ChildObjectIdentifier _rodsRoot;
+        private Constructable _buildable;
+
+        [ProtoMember(3, OverwriteList = true)]
+        [NonSerialized]
+        private CyNukeReactorSaveData _saveData;
+
+        internal readonly TechType[] TechTypePerSlot = new TechType[MaxSlots] { TechType.None, TechType.None, TechType.None, TechType.None };
+        internal readonly float[] ChargePerSlot = new float[MaxSlots] { -1f, -1f, -1f, -1f };
```

Comment:
Consider using a `struct` that combines these two types instead of using parallel arrays.